Sources/SpeziContact/Models/ContactOption.swift

```diff
@@ -38,15 +38,15 @@ public struct ContactOption {
     /// The title representing the `ContactOption` in the user interface.
     public let title: String
     /// The action that should be taken when a user chooses to use the `ContactOption`.
-    public let action: @MainActor () -> Void
+    public let action: any ContactOptionAction
 
     
     /// Encodes a way to get in contact with an individual and usually connected to a `Contact`.
     /// - Parameters:
     ///   - image: The image representing the `ContactOption` in the user interface.
     ///   - title: The title representing the `ContactOption` in the user interface.
     ///   - action: The action that should be taken when a user chooses to use the `ContactOption`.
-    public init(image: Image, title: String, action: @escaping @MainActor () -> Void) {
+    public init(image: Image, title: String, action: any ContactOptionAction) {
```

Comment:
I think we should still keep the closure-base initializer; it would allow us to only tag a minor version increment and might still be useful for a Swift-style API? We could mark it as deprecated in case we don't want to have the same overload for the Android API but at least would make this a non-breaking change.